src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -308,7 +284,45 @@ private void PerformBlockPurge(IDisposable purgeRequest)
                 }
             }
 
+            if (lastMessagePurged.HasValue && OnItemPurged != null)
+            {
+                OnItemPurged(lastMessagePurged.Value);
```

Comment:
I see. Yes, that's what I understood after reviewing more of the code. Please add an xml comment to OnItemPurged that explains this (current) behavior